src/OpenSage.Game/Logic/Object/Update/SlavedUpdate.cs

```diff
@@ -1,8 +1,172 @@
-﻿using System.Numerics;
+﻿using System;
+using System.Numerics;
 using OpenSage.Data.Ini;
+using OpenSage.Graphics.ParticleSystems;
+using OpenSage.Mathematics;
+using OpenSage.Mathematics.FixedMath;
 
 namespace OpenSage.Logic.Object
 {
+    public class SlavedUpdateModule : UpdateModule
+    {
+        private readonly SlavedUpdateModuleData _moduleData;
+        protected readonly GameObject _gameObject;
+        public GameObject Master;
+
+        private double _waitUntil;
```

Comment:
Why not store this as a TimeSpan, then you can do less work below.

Reply:
Good catch! Using TimeSpan now